mesh/mesh.go

```diff
@@ -500,3 +500,66 @@ func (m *Mesh) GetATXs(atxIds []types.AtxId) (map[types.AtxId]*types.ActivationT
 	}
 	return atxs, mIds
 }
+
+// ActiveSetForLayerView - returns the active set size that matches the view of the contextually valid blocks in the provided layer
+func (m *Mesh) ActiveSetForLayerView(layer types.LayerID, layersPerEpoch uint16) (uint32, error) {
```

Comment:
Why re write this function? this is the same code as CalcActiveSet in activation.go
Why?

Reply:
I go over contextually valid blocks only.
I don't count the ATXs from the same miner twice.

We can further abstract the traversalFunction and make the validation block a function parameter for example.

I feel we better live peacefully with this code duplication for know and see where it stabilizes.